pagopa/io-app
# chore: [IOBP-136] Implementation of new lIst item in wallet transactions

## Short description
the wallet's transactions list now displays the correct new DS component, and a placeholder in case no imgLink is provided

<img width="200" alt="Screenshot 2023-07-18 at 16 41 27" src="https://github.com/pagopa/io-app/assets/60693085/1da873e4-7841-437d-9186-de0c0fcb3f90">

## List of changes proposed in this pull request

- refactored `TransactionsList` to be a Functional Component
-`TransactionsList` now renders the correct component
- moved isImageUri from `ListItemTransaction` to `utils/uri`
- removed unused prop in `TransactionsList`

## How to test
navigate to `WALLET_HOME_SCREEN` and load the transaction list, then make sure everything renders correctly and all list items navigate to the transaction's details on press
